float32 x
float32 y
float32 theta
bool is_good_reading